### PR TITLE
Detailed Setup for Windows 10

### PR DESCRIPTION
**Pull request by siddhanth339**
## Summary
Detailed instructions are provided on how to set caMicroscope locally for Windows 10 Home

## Motivation
Getting the codebase running locally is challenging without proper instructions. I wish to make it easy for other developers to get it running locally.

### CODE REVIEW

File: windowsSetup.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+1. Install Windows Subsystem for Linux (WSL2) from the link given here: <a href = "https://docs.microsoft.com/en-us/windows/wsl/install-win10#manual-installation-steps">WSL2</a>, **FOLLOW STEPS 1 to 5 ONLY (Step 6 - Install your Linux distribution of choice IS NOT REQUIRED)** 
+2. After completing all the five steps, <a href = "">Install Docker Desktop for Windows</a>, click on the "Get Docker" button on the right and run the "Docker Desktop Installer" to complete the installation.
+3. Now, select any folder (say E:/) and open the command prompt.
+4. Clone the three repositories: caMicroscope, Distro and Caracal using <code>git clone</code> commands given below:<br>
+    <code>git clone https://github.com/camicroscope/caMicroscope.git</code><br>
+    <code>git clone https://github.com/camicroscope/Caracal.git</code><br>
+    <code>git clone https://github.com/camicroscope/Distro.git</code><br>
+
+5. Open the "caMicroscope.yml" file which can be found in the Distro folder, "Distro/caMicroscope.yml".
+6. Add <code>DISABLE_SEC: "true"</code> as shown below, adding this line will help us get rid of JsonWebTokenError.
```
Comment (birm) [inline windowsSetup.md:12]: This is part of what we use develop.yml for.

File: windowsSetup.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Set up caMicroscope for Windows 10 Home
+
+1. Install Windows Subsystem for Linux (WSL2) from the link given here: <a href = "https://docs.microsoft.com/en-us/windows/wsl/install-win10#manual-installation-steps">WSL2</a>, **FOLLOW STEPS 1 to 5 ONLY (Step 6 - Install your Linux distribution of choice IS NOT REQUIRED)** 
+2. After completing all the five steps, <a href = "">Install Docker Desktop for Windows</a>, click on the "Get Docker" button on the right and run the "Docker Desktop Installer" to complete the installation.
+3. Now, select any folder (say E:/) and open the command prompt.
+4. Clone the three repositories: caMicroscope, Distro and Caracal using <code>git clone</code> commands given below:<br>
+    <code>git clone https://github.com/camicroscope/caMicroscope.git</code><br>
+    <code>git clone https://github.com/camicroscope/Caracal.git</code><br>
```
Comment (birm) [inline windowsSetup.md:8]: Are you doing anything with these modules locally?

Comment (siddhanth339) [reply windowsSetup.md:8]: Not really, to host the application locally, docker-compose only needs the Distro repo (correct me if I am wrong). But I have cloned the caMicroscope repo to see the frontend flow of the application and Caracal for the backend.

File: windowsSetup.md (hunk 1 above)
Comment (siddhanth339) [reply windowsSetup.md:12]: Yes, you are right. Can you please tell me the difference between DISABLE_SEC and ALLOW_PUBLIC? Also, I am new to Docker, so can you tell me the main difference between running caMicroscope.yml file and develop.yml file.

Comment (birm) [reply windowsSetup.md:12]: ALLOW_PUBLIC keeps login checks, but allows someone who has not been signed up to access public slides. See https://github.com/camicroscope/Caracal/blob/2a5be469a1e61fece67cafd2d5bb18c8510b2792/handlers/userFunction.js#L9
DISABLE_SEC ignores all auth checks.

Comment (birm) [reply windowsSetup.md:12]: develop.yml and caMicrosope.yml use different branches and have some config changes to make development easier, and setting DISABLE_SEC is one of those.

Comment (siddhanth339) [reply windowsSetup.md:12]: Thank you.

**Comment by birm** [comment]
I think that this kind of guide should live in camicroscope/distro. Maybe you'd find it useful to change the hosted setup section of the readme? (https://github.com/camicroscope/camicroscope#hosted-setup)
Most of this is not windows specific, except for the advice to use the linux subsystem, which I think could be useful as a tip on the Distro readme.

**Comment by siddhanth339** [comment]
According to your suggestion, I have added a link to the hosted setup which points to the Distro Readme file. 
PR for Distro Readme: https://github.com/camicroscope/Distro/pull/162
PR for hosted setup: https://github.com/camicroscope/caMicroscope/pull/510 (this one)

**Comment by siddhanth339** [comment]
What I feel is all of this should be placed in the Fast local changes section: https://github.com/camicroscope/caMicroscope#fast-local-changes because it is not clear as to which file should be updated and what should exactly be updated for new contributors

**Comment by birm** [comment]
I'd recommend augmenting or clarifying whatever you feel is necessary within the fast local changes without being redundant with our other docs.
Or, we could move the fast local changes section to distro.

**Pull request closed without merging**